compliance/collection/nodescanv2/nodescan.go

```diff
@@ -16,5 +19,60 @@ type NodeScan struct {
 
 // Scan scans the current node and returns the results as storage.NodeInventory object
 func (n *NodeScan) Scan(nodeName string) (*storage.NodeInventory, error) {
-	return nil, errors.New("Not implemented")
+	componentsHost, err := nodes.Analyze(nodeName, "/host/", false)
+	log.Info("Finished node inventory /host scan")
+	if err != nil {
+		log.Errorf("Error scanning node /host inventory: %v", err)
+	}
+	log.Infof("Components found under /host: %v", componentsHost)
+	if err != nil {
+		return nil, err
+	}
+
+	var protoComponents *scannerV1.Components
+	if componentsHost != nil {
+		protoComponents = protoComponentsFromScanComponents(componentsHost)
+	}
+	m := &storage.NodeInventory{
+		NodeName:   nodeName,
+		ScanTime:   timestamp.TimestampNow(),
+		Components: protoComponents,
+	}
+	return m, nil
+}
+
+func protoComponentsFromScanComponents(c *nodes.Components) *scannerV1.Components {
```

Comment:
I need to run now but I'll still leave it here as a question: have you tried researching if Scanner code has functions that do this conversion/mapping? It is unhandy to do this conversion in Compliance code given that something like this may exist in Scanner already.

If it doesn't exist in this form, can some code refactoring be done in Scanner's node analysis so that it produces data in convenient contract for Compliance so that it just takes `scannerV1.Components` data, simply puts into `storage.NodeInventory` and forwards that further?

Reply:
I've asked @RTann this question as well, and the reasoning behind this is as follows:
There is no fitting conversion for this use case in Scanner.
Conversion should be done by the calling parties, not by Scanner itself if possible, to keep Scanner from growing due to many different conversion functions.

Reply:
I was able to find a conversion function from `database.FeatureVersion` to "apiV1".`Feature`.

https://github.com/stackrox/scanner/blob/0316703b6e14713d968157eb8236a9d8659efbef/api/v1/models.go#L74

From there it is convertible to "protoV1".`Feature`.

https://github.com/stackrox/scanner/blob/935c3b337b6ef553d689f55e5a17ea191d90daad/api/v1/imagescan/convert.go#L68

Doesn't look like we need to invent something new. What worries me is that `node.go`'s `Components` exposes `database.*` structs which looks like a wrong level of abstraction to expose.

When will this become a problem? When we put Scanner into own DaemonSet or in a separate container within Collector pods because we'll have to expose node scanning as API and that will require exposing things that are in proto contracts. `database.*` structs won't work because they can't be serialized over gRPC.

Since this `Analyze` API isn't something invented by you, I'll not insist for this issue to be fixed in this PR. Let's have it in mind as something to take care about later.